akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -340,6 +350,50 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
             next(registry.removeNode(uniqueAddress))
           }
 
+        case NodeUnreachable(uniqueAddress) =>
+          val keysForNode = registry.keysFor(uniqueAddress)
+          val newRegistry = registry.addUnreachable(uniqueAddress)
+          if (keysForNode.nonEmpty) {
+            ctx.log.debug(
+              "ClusterReceptionist [{}] - Node with registered services unreachable [{}]",
+              cluster.selfAddress,
+              uniqueAddress)
+            keysForNode.foreach { changedKey =>
+              val serviceKey = changedKey.asServiceKey
+
+              val subscribers = subscriptions.get(changedKey)
+              if (subscribers.nonEmpty) {
+                val (reachable, all) = newRegistry.activeActorRefsFor(serviceKey, selfUniqueAddress)
+                val listing =
+                  ReceptionistMessages.Listing(serviceKey, reachable, all, allServiceInstancesChanged = false)
+                subscribers.foreach(_ ! listing)
+              }
+            }
+          }
+          next(newRegistry)
+
+        case NodeReachable(uniqueAddress) =>
+          val keysForNode = registry.keysFor(uniqueAddress)
+          val newRegistry = registry.removeUnreachable(uniqueAddress)
+          if (keysForNode.nonEmpty) {
+            ctx.log.debug(
+              "ClusterReceptionist [{}] - Node with registered services reachable again [{}]",
+              cluster.selfAddress,
+              uniqueAddress)
+            keysForNode.foreach { changedKey =>
+              val serviceKey = changedKey.asServiceKey
+
+              val subscribers = subscriptions.get(changedKey)
+              if (subscribers.nonEmpty) {
+                val (reachable, all) = newRegistry.activeActorRefsFor(serviceKey, selfUniqueAddress)
+                val listing =
+                  ReceptionistMessages.Listing(serviceKey, reachable, all, allServiceInstancesChanged = false)
+                subscribers.foreach(_ ! listing)
+              }
+            }
+          }
+          next(newRegistry)
+
```

Comment:
Should we also add `UnreachableDataCenter`, which would affect all entries owned by members with the dc?
Could be in separate ticket/pr since it's not as important.

Reply:
We'll get both that and an unreachable event per node right? 

Let's ticketify as future improvement.

Reply:
https://github.com/akka/akka/issues/27327